test/unit/update_and_kill_test.lua

```diff
@@ -78,3 +78,38 @@ function g.test_expirationd_update(cg)
         expirationd.kill("test" .. i)
     end
 end
+
+function g.test_zombie_task_kill(cg)
+    local space = cg.space
+
+    local one_hour = 3600
+    local task = expirationd.start("test", space.id, helpers.is_expired_true,
+            {
+                full_scan_delay = one_hour,
+            })
+
+    local first_task_fiber
+    helpers.retrying({}, function()
+        first_task_fiber = task.worker_fiber
+        t.assert_equals(first_task_fiber:status(), "suspended")
+    end)
+    local total = 10
+    for i = 1, total do
+        space:insert({i, tostring(i)})
+    end
+    fiber.sleep(3)
```

Comment:
Why do you need a sleep here?

Reply:
To show for sure that even after 3 seconds, nothing will change and the fiber will sleep, I agree that the solution is not the best.